lib/reverse_sentence.rb

```diff
@@ -1,6 +1,89 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
+
 def reverse_sentence(my_sentence)
```

Comment:
This works and you have your BigO predicted correctly. However there is an alternative solution which is also O(n) time complexity and O(1) space complexity.

Consider: Reverse the string completely. Then loop through the string reversing each individual word, in place (i.e. not making an array or another string). Think about how you might be able to do that.